icu4c/source/common/locdispnames.cpp

```diff
@@ -412,6 +409,7 @@ uloc_getDisplayLanguage(const char *locale,
                         const char *displayLocale,
                         char16_t *dest, int32_t destCapacity,
                         UErrorCode *pErrorCode) {
+    if (U_FAILURE(*pErrorCode)) { return 0; }
```

Comment:
optional: a few of these functions only call one other function each which takes a UErrorCode